src/ui/public/index_patterns/_field_format/content_types.js

```diff
@@ -1,73 +1,69 @@
-          return sub.indexOf('\n') > -1;
-        });
-
-        return subVals.join(',' + (useMultiLine ? '\n' : ' '));
+const types = {
+  html: function (format, convert) {
+    function recurse(value, field, hit) {
+      if (value == null) {
+        return _.asPrettyString(value);
       }
 
-      return function (...args) {
-        return `<span ng-non-bindable>${recurse(...args)}</span>`;
-      };
-    },
+      if (!value || typeof value.map !== 'function') {
+        return convert.call(format, value, field, hit);
+      }
 
-    text: function (format, convert) {
-      return function recurse(value) {
-        if (!value || typeof value.map !== 'function') {
-          return convert.call(format, value);
-        }
+      const subVals = value.map(function (v) {
+        return recurse(v, field, hit);
+      });
+      const useMultiLine = subVals.some(function (sub) {
+        return sub.indexOf('\n') > -1;
+      });
 
-        // format a list of values. In text contexts we just use JSON encoding
-        return angular.toJson(value.map(recurse), true);
-      };
+      return subVals.join(',' + (useMultiLine ? '\n' : ' '));
     }
-  };
 
-  function fallbackText(value) {
-    return _.asPrettyString(value);
-  }
+    return function (...args) {
+      return `<span ng-non-bindable>${recurse(...args)}</span>`;
+    };
+  },
 
-  function fallbackHtml(value, field, hit) {
-    const formatted = _.escape(this.convert(value, 'text'));
+  text: function (format, convert) {
+    return function recurse(value) {
+      if (!value || typeof value.map !== 'function') {
+        return convert.call(format, value);
+      }
 
-    if (!hit || !hit.highlight || !hit.highlight[field.name]) {
-      return formatted;
-    } else {
-      return getHighlightHtml(formatted, hit.highlight[field.name]);
-    }
+      // format a list of values. In text contexts we just use JSON encoding
+      return JSON.stringify(value.map(recurse));
+    };
   }
+};
 
-  function setup(format) {
-    const src = format._convert || {};
-    const converters = format._convert = {};
+function fallbackText(value) {
+  return _.asPrettyString(value);
+}
 
-    converters.text = types.text(format, src.text || fallbackText);
-    converters.html = types.html(format, src.html || fallbackHtml);
+function fallbackHtml(value, field, hit) {
+  const formatted = _.escape(this.convert(value, 'text'));
 
-    return format._convert;
+  if (!hit || !hit.highlight || !hit.highlight[field.name]) {
+    return formatted;
+  } else {
+    return getHighlightHtml(formatted, hit.highlight[field.name]);
   }
+}
 
-  return {
-    types: types,
-    setup: setup
-  };
+function setup(format) {
+  const src = format._convert || {};
+  const converters = format._convert = {};
+
+  converters.text = types.text(format, src.text || fallbackText);
+  converters.html = types.html(format, src.html || fallbackHtml);
+
+  return format._convert;
 }
+
+export const contentTypes = {
```

Comment:
Is there a concern of naming collisions? Should `setup` and `types` get some kind of prefix to make them more unique?